### DISCUSSION TITLE
a!1 in the smt output

### DISCUSSION THREAD

**Discussion by Nivous** [seahorn/seahorn#552] [Q&A]
for some reason I am getting some rules containing this expression: a!1. Any idea what does it mean? When is it being added? I am not sure what is the purpose here

**Comment by Nivous** [comment]
Any help would be appreciated

**Comment by agurfinkel** [comment]
A complete example would help. In general, `a!1` is a temporary symbol introduced by z3 during printing for intermediate let bindings. It is expected and desired.
